rts/lra/jaxrs/src/main/java/io/narayana/lra/filter/ServerLRAFilter.java

```diff
@@ -240,41 +286,65 @@ public void filter(ContainerRequestContext containerRequestContext) {
 
                         // if there is an LRA present nest a new LRA under it
                         suspendedLRA = incommingLRA;
-                        lraTrace(suspendedLRA, "ServerLRAFilter before: REQUIRED start new LRA");
-                        newLRA = lraId = startLRA(incommingLRA, method, timeout);
+
+                        if (progress == null) {
```

Comment:
Why this check for null is needed when the `updateProgress` checks on `null`?

Reply:
Because it might be null and `updateProgress` might not have been called.

Reply:
I could rewrite the whole thing in many different ways. Code is an art as well as a science.